src/webaudio.js

```diff
@@ -196,6 +196,45 @@ WaveSurfer.WebAudio = {
 
         return peaks;
     },
+    
+	/**
+     * @returns {Float32Array} Array of peaks for both channel.
+     */
+    getPeaksMainWaveform: function (length) {
+        var buffer = this.buffer;
+        var sampleSize = buffer.length / length;
+        var sampleStep = ~~(sampleSize / 10) || 1;
+        var channels = buffer.numberOfChannels;
+        var peaks = new Float32Array(channels*length);
+
+        for (var c = 0; c < channels; c++) {
+            var chan = buffer.getChannelData(c);
+            for (var i = 0; i < length; i++) {
+                var start = ~~(i * sampleSize);
+                var end = ~~(start + sampleSize);
+                var max = 0;
+                for (var j = start; j < end; j += sampleStep) {
+                    var value = chan[j];
+                    if (value > max) {
+                        max = value;
+                    // faster than Math.abs
+                    } else if (-value > max) {
+                        max = -value;
+                    }
+                }
+                
+				//Sequentially
+				peaks[c*length+i] = max;
```

Comment:
I don't like the idea of push all channels into the same flat array. I'd rather go for an array of arrays:

    [ [ 0.5, ..., 0.8 ], [ 0.1, ..., 0.4 ] ]

To avoid breaking things, one could still accept a flat array (for mono) *and* an array of arrays (for stereo and higher).